crates/sui-config/src/node.rs

```diff
@@ -179,6 +179,19 @@ pub struct NodeConfig {
 
     #[serde(skip_serializing_if = "Option::is_none")]
     pub firewall_config: Option<RemoteFirewallConfig>,
+
+    #[serde(default)]
+    pub execution_cache: ExecutionCacheConfig,
+}
+
+#[derive(Clone, Debug, Deserialize, Serialize, Default)]
+#[serde(rename_all = "kebab-case")]
+pub enum ExecutionCacheConfig {
```

Comment:
Not sure what you mean - assuming the cache is working correctly, there is no need to synchronize the process of validators enabling it, because it isn't a protocol change. They can turn it on at any time.

I also believe that it should be safe to switch back and forth, but I will need to test that unless we want to encourage people not to do that.